core/core.go

```diff
@@ -145,10 +150,10 @@ func (c *Core) procAppendQueue() {
 	for _, hash := range c.appendQueue.Keys() {
 		if value, exist := c.appendQueue.Peek(hash); exist {
 			hashNumber := types.HashAndNumber{Hash: hash.(common.Hash), Number: value.(blockNumberAndRetryCounter).number}
-			if value.(blockNumberAndRetryCounter).retry < c_appendQueueRetryPriorityThreshold {
-				hashNumberPriorityList = append(hashNumberPriorityList, hashNumber)
-			}
 			if hashNumber.Number < c.CurrentHeader().NumberU64()+c_maxFutureBlocks {
+				if value.(blockNumberAndRetryCounter).retry < c_appendQueueRetryPriorityThreshold {
+					hashNumberPriorityList = append(hashNumberPriorityList, hashNumber)
+				}
 				hashNumberList = append(hashNumberList, hashNumber)
```

Comment:
Should this be an else? do we want entries in both priority and number list at the same time?

Reply:
Put it in the else